pkg/states/state_manager.go

```diff
@@ -0,0 +1,77 @@
+package states
+
+import "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+func Active(userSignup *v1alpha1.UserSignup) bool {
+	return Approved(userSignup) &&
+		!VerificationRequired(userSignup) &&
+		!Deactivated(userSignup)
+}
+
+func Approved(userSignup *v1alpha1.UserSignup) bool {
+	return contains(userSignup.Spec.States, v1alpha1.UserSignupStateApproved)
+}
+
+func SetApproved(userSignup *v1alpha1.UserSignup, val bool) {
+	setState(userSignup, v1alpha1.UserSignupStateApproved, val)
+
+	if val {
+		setState(userSignup, v1alpha1.UserSignupStateVerificationRequired, false)
```

Comment:
how about also setting:
```
setState(userSignup, v1alpha1.UserSignupStateDeactivated, false)
```
I guess that we should make sure that the deactivated state is not present when approving a user.

Reply:
@sbryzak WDYT about this suggestion ^^^?

Reply:
Good idea, I'll be adding a subtask for this in the following story: https://issues.redhat.com/browse/CRT-1003